components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go

```diff
@@ -175,6 +176,288 @@ func TestReconciler_Reconcile(t *testing.T) {
+			}
+
+			apiRule := &apigatewayv1beta1.APIRule{}
+			err = cli.Get(ctx, namespacedName, apiRule)
+			require.NoError(t, err)
+
+			// then
+			require.Equal(t, tc.wantHandler.Name, apiRule.Spec.Rules[0].AccessStrategies[0].Handler.Name)
+			require.Equal(t, tc.wantHandler.Config, apiRule.Spec.Rules[0].AccessStrategies[0].Handler.Config)
+		})
+	}
+}
+
+// TestReconciler_APIRuleConfig_Upgrade ensures that the created APIRule is configured correctly
+// before and after the upgrade from ory to Eventing webhook auth and vise versa.
+func TestReconciler_APIRuleConfig_Upgrade(t *testing.T) {
+	ctx := context.Background()
+
+	credentials := &eventmesh.OAuth2ClientCredentials{
+		CertsURL: "https://domain.com/oauth2/certs",
+	}
+
+	subscription := reconcilertesting.NewSubscription("some-test-sub", "test",
+		reconcilertesting.WithDefaultSource(),
+		reconcilertesting.WithValidSink("test", "some-test-svc"),
+		reconcilertesting.WithFinalizers([]string{eventingv1alpha2.Finalizer}),
+		reconcilertesting.WithEventType(reconcilertesting.OrderCreatedEventType),
+		reconcilertesting.WithConditions(eventingv1alpha2.MakeSubscriptionConditions()),
+		reconcilertesting.WithEmsSubscriptionStatus(string(types.SubscriptionStatusActive)),
+	)
+
+	validator := sink.ValidatorFunc(func(s *eventingv1alpha2.Subscription) error { return nil })
+
+	var testCases = []struct {
+		name                            string
+		givenSubscription               *eventingv1alpha2.Subscription
+		givenReconcilerSetup            func() (*Reconciler, client.Client)
+		givenEventingWebhookAuthEnabled bool
+		wantReconcileResult             ctrl.Result
+		wantReconcileError              error
+		wantHandlerBeforeUpgrade        apigatewayv1beta1.Handler
+		wantHandlerAfterUpgrade         apigatewayv1beta1.Handler
+	}{
+		{
+			name:              "Eventing webhook auth is not enabled before the upgrade",
+			givenSubscription: subscription,
+			givenReconcilerSetup: func() (*Reconciler, client.Client) {
+				te := setupTestEnvironment(t, subscription)
+				te.credentials = credentials
+				te.backend.On("Initialize", mock.Anything).Return(nil)
+				te.backend.On("SyncSubscription", mock.Anything, mock.Anything, mock.Anything).Return(true, nil)
+				return NewReconciler(ctx, te.fakeClient, te.logger, te.recorder, te.cfg, te.cleaner, te.backend,
+					te.credentials, te.mapper, validator), te.fakeClient
+			},
+			givenEventingWebhookAuthEnabled: false,
+			wantReconcileResult:             ctrl.Result{},
+			wantReconcileError:              nil,
+			wantHandlerBeforeUpgrade: apigatewayv1beta1.Handler{
+				Name:   object.OAuthHandlerNameOAuth2Introspection,
+				Config: nil,
+			},
+			wantHandlerAfterUpgrade: apigatewayv1beta1.Handler{
+				Name: object.OAuthHandlerNameJWT,
+				Config: &runtime.RawExtension{
+					Raw: []byte(fmt.Sprintf(object.JWKSURLFormat, credentials.CertsURL)),
+				},
+			},
+		},
+		{
+			name:              "Eventing webhook auth is enabled before the upgrade",
```

Comment:
please see the [comment](https://github.com/kyma-project/kyma/pull/17632/files#r1229571234)

Reply:
Please check https://github.com/kyma-project/kyma/pull/17632#discussion_r1229603629